### ISSUE TITLE
Simplify/merge function modules

### ISSUE THREAD

**Issue by Sir4ur0n** [c-cube/qcheck#124]
Currently there are at least 4 modules related to function generation
* `Fn`
* `Tuple`
* `Observable`
* `QCheck` (top level)

While there **may** be room for reuse of `Tuple` for other things (e.g. for [curried functions](https://github.com/c-cube/qcheck/issues/108)) I think it's too much unnecessary burden to read and grasp through 4 modules to get one feature (function generator)

Note also that if `Tuple` remains standalone (e.g. for curried functions) then I think `Observable` related stuff like `obs` and `o_nil` need to be removed/moved somewhere else.

I don't have a clear view of what's ideal, but I suspect the API can be improved for users

**Comment by c-cube** [comment]
The API is definitely streamlinable, I hope it can be done once we have integrated shrinking ready :)

**Comment by c-cube** [comment]
I think this is done in #110 and later?

**Comment by Sir4ur0n** [comment]
No, I kept all existing modules, so the UX problem remains I think
